### PR TITLE
Fix typos & add pre-commit configuration

### PR DESCRIPTION
**Pull request by akx**
The pre-commit configuration was adapted from Ruff's.

Typos needed to be fixed because they'd fail pre-commit otherwise now. 😁

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    .*/(snapshots)/.*|
+  )$
+
+repos:
+  - repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.16
+    hooks:
+      - id: validate-pyproject
+
+  - repo: https://github.com/crate-ci/typos
```
Comment (MichaReiser) [inline .pre-commit-config.yaml:14]: No ruff? ;)

Comment (akx) [reply .pre-commit-config.yaml:14]: Oh, fine, this time, if you insist... 😂

**Review by zanieb — APPROVED** [review]
Sweet!

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+exclude: |
+  (?x)^(
+    .*/(snapshots)/.*|
+  )$
+
+repos:
+  - repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.16
+    hooks:
+      - id: validate-pyproject
```
Comment (zanieb) [inline .pre-commit-config.yaml:12]: Do we need this one?

Comment (zanieb) [reply .pre-commit-config.yaml:12]: I think Ruff has support for this?

Comment (henryiii) [reply .pre-commit-config.yaml:12]: Validate-pyproject validates schemas. Ruff just makes sure the pyproject is parsable, as far as I know, and checks its own section. It doesn't check schemas (yet?).

It's not that useful though unless you add the schemas from SchemaStore, which can be done either with --store (slow, pulls from network) or `validate-pyproject-schema-store`, which is a regularly updated copy of the pyproject related schemas. Currently about half of the most popular projects are supported with schemas: https://github.com/SchemaStore/schemastore/issues/3564.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    hooks:
+      - id: prettier
+        types: [yaml]
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.2.1
+    hooks:
+      - id: ruff-format
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
```
Comment (henryiii) [inline .pre-commit-config.yaml:39]:
```suggestion
        args: [--fix, --exit-non-zero-on-fix, --show-fixes]
```

Also, I believe I've been told `ruff-format` should follow `ruff`.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (akx) [reply .pre-commit-config.yaml:12]: Yeah, this was literally just copied from Ruff :D

File: .pre-commit-config.yaml (hunk 3 above)
Comment (akx) [reply .pre-commit-config.yaml:39]: Also copied from Ruff, no opinions either way.

**Review by MichaReiser — APPROVED** [review]

**Pull request merged**